File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost) {
        int n = cost.length;
        Arrays.sort(cost);
        int i = n-1, ans = 0, cnt = 0;
        while(i > -1){
            if(cnt < 2){
                ans += cost[i];
                cnt++;
            }
            else{
                cnt = 0;
            }
            i--;
        }
        return ans;
    }
}

File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minimumCost = function (cost) {
    let n = cost.length;
    cost.sort(function(a,b){return a-b});
    let i = n - 1, ans = 0, cnt = 0;
    while (i > -1) {
        if (cnt === 2) {
            cnt = 0;
            i--;
            continue;
        }
        ans += cost[i];
        cnt++;
        i--;
    }
    return ans;
};